lib/modules/platform/github/list-cache.ts

```diff
@@ -0,0 +1,65 @@
+import { DateTime } from 'luxon';
+import type { RestPageCache, RestPageItem } from './types';
+
+export function getEmptyCache<T extends RestPageItem>(): RestPageCache<T> {
+  return {
+    items: {},
+    timestamp: DateTime.fromISO('1900-01-01').toISO(),
+    etag: '',
+  };
+}
+
+export function getItem<T extends RestPageItem>(
```

Comment:
i think this is a good candidate for a class

Reply:
My point is to attach the cache object to repo cache structure during init phase to mutate it during run:

https://github.com/zharinov/renovate/blob/8391721640901be24b78d67af36ed8122d3abd71/lib/util/cache/repository/types.ts#L56

The finalize phase is unavailable from the platform code, so it seems like we don't have enough control to class serialization/deserialization.

Reply:
we could wrap it when getting from cache, so object ref is still hold?

Reply:
Mm, now I get it, good idea 👍